feature: Sprites and animations for the player and enemies

pablo-zucchi opened the pull request.

pablo-zucchi committed:
feature: Sprites and animations for the player and enemies

Paaprikaa commented on Assets/Scripts/Enemy.cs:
```diff
@@ -1,88 +1,122 @@
+        _nextAttackTime = _attackCooldown;
+    }
+
+    public override void Death()
```
Forgot to add "Destroy(gameObject)"

pablo-zucchi replied:
Added again, will be removed when death animation is done.

Paaprikaa commented on Assets/Scripts/Enemy.cs:
```diff
@@ -1,88 +1,122 @@
+        _animator = GetComponent<Animator>();
     }
 
     private void Update()
```
Please move everything to OnUpdate(). Update function **should** look always like this:
private void Update() { OnUpdate(); }

--> This is a good practice to avoid problems form children classes

Paaprikaa commented on Assets/Scripts/PlayerMovement.cs:
```diff
@@ -6,20 +6,27 @@ public class PlayerMovement : MonoBehaviour
     private void Update()
     {
+
+        if (!canMove) { _rb.linearVelocity = Vector2.zero; return; }
```
There is some repeated code here, not necessary to assign the velocity to he rb in Update(), it's okey that is only assigned in the FixedUpdate() (since it's for physics).

Paaprikaa commented on Assets/Scripts/RangedEnemy.cs:
```diff
@@ -1,18 +1,33 @@
+        );
+    }
+
+    public void OnAttackShoot()
```
Why is Attack() not being directly called?

pablo-zucchi replied:
It was a mistake, now attack is directly called.

Paaprikaa commented on Assets/Sprites/modules/house_back.png:
If you have sprites called for example "shed_front_1", then change "shed_front" to "shed_front_0", or start counting from 1 if you wish.

Paaprikaa reviewed (changes requested):
Please check all comments, ask if you have any question

pablo-zucchi commented on Assets/Scripts/Enemy.cs:
```diff
@@ -1,88 +1,122 @@
-                _nextAttackTime = _attackCooldown;
-            }
-        } else
+        if (!_onAggro)
```
Fixed. Now velocity is always zero if the enemy isn´t moving, has no target or isn't in aggro mode.

pablo-zucchi commented on Assets/Scripts/RangedEnemy.cs:
```diff
@@ -1,18 +1,33 @@
-            // See preferences.
-            GameObject bullet = BulletPool.Instance.RequestBullet(_firingPoint.transform.position, transform.rotation);
-        }
+        if (Vector2.Distance(target.position, transform.position) > distanceToShoot)
```
Done.

pablo-zucchi committed:
bugfix: Requested Changes

Paaprikaa commented on Assets/Scripts/MeleeEnemy.cs:
```diff
@@ -2,24 +2,30 @@
 public class MeleeEnemy : Enemy
 {
-    [SerializeField] private float _hitRadius;
+    [SerializeField] private float _hitRadius = 1f;
```
Still, 1.6f is not enough to exceed the sprite. But there is no problem, it'll be changed later when we test what "feels" better in game

pablo-zucchi committed:
bugfix: Ranged Attack didn't work

I reverted the change in the last commit, this line may seem redundant but its necessary for the ranged attack, without it, it doesn't work
